File: bindings/php/src/ProvenIds.php
<?php
// SPDX-License-Identifier: PMPL-1.0-or-later
// Copyright (c) 2026 Jonathan D.A. Jewell (hyperpolymath) <j.d.a.jewell@open.ac.uk>
//
// IDS protocol types for proven-servers.

declare(strict_types=1);

namespace ProvenServers;

/** AlertSeverity matching the Idris2 ABI tags. */
enum AlertSeverity: int
{
    case AlertSeverity_Low = 0;
    case AlertSeverity_Medium = 1;
    case AlertSeverity_High = 2;
    case AlertSeverity_Critical = 3;
}

/** DetectionMethod matching the Idris2 ABI tags. */
enum DetectionMethod: int
{
    case Signature = 0;
    case Anomaly = 1;
    case Stateful = 2;
    case Heuristic = 3;
}

/** IdsProtocol matching the Idris2 ABI tags. */
enum IdsProtocol: int
{
    case Tcp = 0;
    case Udp = 1;
    case Icmp = 2;
    case Dns = 3;
    case Http = 4;
    case Tls = 5;
    case Ssh = 6;
}

/** IdsAction matching the Idris2 ABI tags. */
enum IdsAction: int
{
    case Alert = 0;
    case Drop = 1;
    case Log = 2;
    case Block = 3;
    case Pass = 4;
}

/** Direction matching the Idris2 ABI tags. */
enum Direction: int
{
    case Inbound = 0;
    case Outbound = 1;
    case Both = 2;
}

/** ThreatLevel matching the Idris2 ABI tags. */
enum ThreatLevel: int
{
    case Info = 0;
    case ThreatLevel_Low = 1;
    case ThreatLevel_Medium = 2;
    case ThreatLevel_High = 3;
    case ThreatLevel_Critical = 4;
}

File: bindings/csharp/src/ProvenIds.cs
// SPDX-License-Identifier: MPL-2.0
// Copyright (c) 2026 Jonathan D.A. Jewell (hyperpolymath) <j.d.a.jewell@open.ac.uk>
//
// IDS protocol types for proven-servers.

namespace Proven;

/// <summary>AlertSeverity matching the Idris2 ABI tags (0-3).</summary>
public enum AlertSeverity : byte
{
    Low = 0,
    Medium = 1,
    High = 2,
    Critical = 3
}

/// <summary>DetectionMethod matching the Idris2 ABI tags (0-3).</summary>
public enum DetectionMethod : byte
{
    Signature = 0,
    Anomaly = 1,
    Stateful = 2,
    Heuristic = 3
}

/// <summary>IdsProtocol matching the Idris2 ABI tags (0-6).</summary>
public enum IdsProtocol : byte
{
    Tcp = 0,
    Udp = 1,
    Icmp = 2,
    Dns = 3,
    Http = 4,
    Tls = 5,
    Ssh = 6
}

/// <summary>IdsAction matching the Idris2 ABI tags (0-4).</summary>
public enum IdsAction : byte
{
    Alert = 0,
    Drop = 1,
    Log = 2,
    Block = 3,
    Pass = 4
}

/// <summary>Direction matching the Idris2 ABI tags (0-2).</summary>
public enum Direction : byte
{
    Inbound = 0,
    Outbound = 1,
    Both = 2
}

/// <summary>ThreatLevel matching the Idris2 ABI tags (0-4).</summary>
public enum ThreatLevel : byte
{
    Info = 0,
    Low = 1,
    Medium = 2,
    High = 3,
    Critical = 4
}
